sympy/sets/sets.py

```diff
@@ -1765,6 +1765,16 @@ class FiniteSet(Set, EvalfMixin):
     is_empty = False
     is_finite_set = True
 
+    def factor(self, *args, **kwargs):
+        """
+        Implements the SymPy factor routine, on the scalar parts
+        of a basis-dependent expression.
+        factor's documentation
+        ========================
+        """
+        fctr_components = [fctr(v, *args, **kwargs) * k for k, v in self.components.items()]
+        return self._add_func(*fctr_components)
+    
```

Comment:
You have trailing whitespace look at [this](https://travis-ci.org/sympy/sympy/jobs/635934071?utm_medium=notification&utm_source=github_status)